src/xxd/xxd.c

```diff
@@ -844,8 +888,61 @@ main(int argc, char *argv[])
       c = addrlen + 1 + (grplen * x) / octspergrp;
       if (hextype == HEX_NORMAL || hextype == HEX_LITTLEENDIAN)
 	{
+          if (color)
+            {
+            COLOR_PROLOGUE
+
+            if (ebcdic)  /* EBCDIC */
```

Comment:
This code is repeated a few lines below, consider introducing a lookup table for converting the EBCDIC/ASCII codepoint to color.

Reply:
I reimplemented it with function.